src/main/java/com/atlassian/db/replica/internal/state/ConnectionState.java

```diff
@@ -189,17 +189,15 @@ private Connection prepareReadConnection(RouteDecisionBuilder decisionBuilder) t
             decisionBuilder.cause(writeConnection.getFirstCause().build());
             return writeConnection.get(decisionBuilder);
         }
-        if (consistency.isConsistent(() -> readConnection.get(decisionBuilder))) {
+        chief.overrideDecision(decisionBuilder, singleton(() -> readConnection.get(decisionBuilder)));
+        if (decisionBuilder.build().willRunOnMain()) {
+            isMainCommited = true;
+            return prepareMainConnection(decisionBuilder);
+        } else {
             if (getState().equals(COMMITED_MAIN)) {
                 closeConnection(writeConnection, decisionBuilder);
             }
-            final Connection connection = readConnection.get(decisionBuilder);
-            replicaConsistent = true;
-            return connection;
-        } else {
-            replicaConsistent = false;
-            decisionBuilder.reason(REPLICA_INCONSISTENT);
-            return prepareMainConnection(decisionBuilder);
+            return readConnection.get(decisionBuilder);
```

Comment:
The `if` builds the decision, but here we're still passing the partial one. Is that intentional?

Reply:
I need to verify it.